prow/plugins/trigger/generic-comment.go

```diff
@@ -132,6 +133,12 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 	if needsHelp, note := pjutil.ShouldRespondWithHelp(gc.Body, len(toTest)); needsHelp {
 		return addHelpComment(c.GitHubClient, gc.Body, org, repo, pr.Base.Ref, pr.Number, presubmits, gc.HTMLURL, commentAuthor, note, c.Logger)
 	}
+	// we want to be able to track re-tests separately from the general body of tests
+	if pjutil.RetestRe.MatchString(gc.Body) || pjutil.RetestRequiredRe.MatchString(gc.Body) {
+		for _, item := range toTest {
```

Comment:
This has three issues:
* You are manipulating the iterator variable, which will not get propagated outside the loop
* Once you fix that, you are going to directly manipulate the loaded config instead of doing a change scoped to this function
* Labels might be nil in which case this will panic